src/dom-parts/ChildNodePart.js

```diff
@@ -304,38 +261,42 @@ export class ChildNodePart extends Part {
         }
         if (Array.isArray(value)) {
             for (let index = 0; index < value.length; index++) {
-                this.parts[index]?.update(value[index]);
+                this.arrayParts[index]?.update(value[index]);
             }
         }
     }
 
     /**
      * @param {TemplateResult | any[] | any} value
-     * @return {TemplatePart | any[] | any}
+     * @return {TemplatePart | Node | string | Array<TemplatePart | Node | string>}
      */
     parseValue(value) {
         if (value instanceof TemplateResult) {
             if (!this.templatePart) {
-                const startNode = Array.from(this.childNodes)
-                    .filter((node) => node.nodeType === COMMENT_NODE)
-                    .find((node) => node.data === 'template-part');
-                this.templatePart = new TemplatePart(startNode, value);
-                const serverSideRendered = startNode !== undefined;
-                if (!serverSideRendered) {
-                    // TODO: we should already have the "endNode" from the constructor...
-                    const endNode = Array.from(this.childNodes)
-                        .filter((node) => node.nodeType === COMMENT_NODE)
-                        .find((node) => node.data.includes('/dom-part-'));
+                this.templatePart = new TemplatePart(this.markers?.findNestedStartNode('template-part'), value);
+                if (!this.markers?.serverSideRendered) {
                     // INFO: this is similar to TemplateResult#237
-                    insertNodesBetweenComments(endNode, this.templatePart.childNodes);
+                    replaceNodesBetweenComments(this.markers?.endNode, this.templatePart.childNodes);
                 }
+            } else if (this.templatePart.strings !== value.strings) {
+                // this means we have a different TemplateResult here with maybe different parts also
+                this.templatePart = new TemplatePart(undefined, value);
```

Comment:
so we dont have a startnode here ? 
Is this intentional ?

Reply:
Yes! Because otherwhise the TemplatePart would assume being rendered serverside and only hydrate itself. But we need to really create a new Part since the strings have changed here. It is not very intuitive I know...